src/golang/cmd/server/handler/delete_workflow.go

```diff
@@ -183,26 +193,65 @@ func (h *DeleteWorkflowHandler) Perform(ctx context.Context, interfaceArgs inter
 			if err != nil {
 				return resp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred while validating objects.")
 			}
-			// No operator had touched the object at the specified integration.
+
+			// No operator had touched the object at the specified integration. However, this does not mean that the object is
+			// invalid. We need to consider the case where the object was part of a parameterized saved.
+			var touchedOperatorLoadParams []connector.LoadParams
 			if len(touchedOperators) == 0 {
-				return resp, http.StatusBadRequest, errors.New("Object list not valid. Make sure all objects are touched by the workflow.")
+				// Fetch and cache this map once for performance.
+				if saveOpsByIntegrationName == nil {
+					saveOpsByIntegrationName = make(map[string][]views.LoadOperator)
+					saveOpsList, err := GetDistinctSaveOpsByWorkflow(
+						ctx,
+						args.WorkflowID,
+						h.OperatorRepo,
+						h.DagRepo,
+						h.ArtifactResultRepo,
+						h.Database,
+					)
```

Comment:
There may be an argument for using `GetDistinctSaveOpsByWorkflow` from the start, instead of going through `h.OperatorRepo.GetLoadOpsByWorkflowAndIntegration()` and then falling back to this method, but I don't understand the difference super well and didn't want to rock the boat too much. cc: @eunice-chan

Reply:
I think you should be able to use `GetDistinctSaveOpsByWorkflow` from the get-go. For consistency's sake it might be be good to call save ops, load ops.

Reply:
Updated. Would appreciate another look! In particular I added a new `NonRelational` cast.